### PR TITLE
Gh 2761 federatedstore federated operation merge mapping

### LINKED ISSUE #2761: FederatedStore: default Merge Mapping
continuation of  #2357

Add a default mappings of what merge function to be used for each operation.

This is should include a mapping for getElements which re-applys views. this defeats an issue with:
```
graphA with entityA count 6
graphB with entityA count 11
graphC with entityB count 7
graphD with entityC count 4

getAllElements with view count less than 10

result should be empty
```

### PR DESCRIPTION
**Pull request by GCHQDev404**
# Related Issue

- Resolve #2761

### CODE REVIEW

File: core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/HasTraitHandler.java [hunk 1]
```diff
@@ -34,6 +34,7 @@ public class HasTraitHandler implements OutputOperationHandler<HasTrait, Boolean
     public Boolean doOperation(final HasTrait operation, final Context context, final Store store) throws OperationException {
         Map<String, String> filteredOptions = operation.getOptions().entrySet()
                 .stream()
+                //TODO FS This Handler has FederatedStore logic.
```
Comment (GCHQDeveloper314) [inline core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/HasTraitHandler.java:37]: Do you mean that ideally there shouldn't be fed store logic in here?

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedStore.java [hunk 2]
```diff
@@ -16,6 +16,8 @@
 
 package uk.gov.gchq.gaffer.federatedstore;
 
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import org.apache.htrace.fasterxml.jackson.annotation.JsonCreator;
```
Comment (GCHQDeveloper314) [inline store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedStore.java:20]: This is the wrong import. This is a shaded version from a different JAR and we cannot control the version.
```suggestion
import com.fasterxml.jackson.annotation.JsonCreator;
```

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/util/MergeSchema.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017-2022 Crown Copyright
```
Comment (GCHQDeveloper314) [inline store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/util/MergeSchema.java:2]: Work started on this in 2017?

File: store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreSchemaTest.java [hunk 4]
```diff
@@ -135,6 +136,7 @@ public void shouldGetCorrectDefaultViewForAChosenGraphOperation() throws Operati
     }
 
     @Test
+    @Disabled //TODO FS OverlappingSchema review
```
Comment (GCHQDeveloper314) [inline store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreSchemaTest.java:139]: Is there a ticket for this/will it be a future alpha?

**Review by GCHQDeveloper314 — CHANGES_REQUESTED** [review]
Minor comments. Only changed required is to fix shaded import.

File: store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreSchemaTest.java (hunk 4 above)
Comment (GCHQDev404) [reply store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreSchemaTest.java:139]: @t92549  is on the case.

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/util/MergeSchema.java (hunk 3 above)
Comment (GCHQDev404) [reply store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/util/MergeSchema.java:2]: no.

File: core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/HasTraitHandler.java (hunk 1 above)
Comment (GCHQDev404) [reply core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/HasTraitHandler.java:37]: yes

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/util/MergeSchema.java (hunk 3 above)
Comment (GCHQDev404) [reply store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/util/MergeSchema.java:2]: No but I think it was duplicated from an earlier class.

**Review by GCHQDeveloper314 — APPROVED** [review]

**Review by t92549 — APPROVED** [review]

**Pull request merged**